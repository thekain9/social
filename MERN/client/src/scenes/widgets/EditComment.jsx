import React, { useState } from 'react';

// Define the EditComment component
function EditComment({ comment, onEdit }) {
  const [editedComment, setEditedComment] = useState(comment.text);
// Handle the edit action and call the onEdit function with updated data
  const handleEdit = () => {
    onEdit(comment.id, editedComment);
  };

  return (
    <div>
      <input value={editedComment} onChange={(e) => setEditedComment(e.target.value)} />
      <button onClick={handleEdit}>Save</button>
    </div>
  );
}

export default EditComment;