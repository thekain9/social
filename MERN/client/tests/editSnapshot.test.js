import React from 'react';
import renderer from 'react-test-renderer';
import EditComment from '../src/scenes/widgets/EditComment';

describe('EditComment Snapshot Test', () => {
  it('matches the snapshot', () => {
    const comment = {
      id: 1,
      text: 'Original comment text',
    };

    const onEdit = jest.fn();

    const tree = renderer.create(
      <EditComment comment={comment} onEdit={onEdit} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
