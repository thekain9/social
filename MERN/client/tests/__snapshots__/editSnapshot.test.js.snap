// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditComment Snapshot Test matches the snapshot 1`] = `
<div>
  <input
    onChange={[Function]}
    value="Original comment text"
  />
  <button
    onClick={[Function]}
  >
    Save
  </button>
</div>
`;
